File: 05- Hash Map - Set/C#/2- Unique Number of Occurrences.cs
public class Solution {
    public bool UniqueOccurrences(int[] arr) {
        Dictionary<int , int> freq = new Dictionary<int, int>() ; 

        foreach(int x in arr){
            if (freq.ContainsKey(x))
                freq[x]++ ; 
            else 
                freq[x] = 1 ; 
        }
            
        HashSet<int> st = new HashSet<int>() ; 
        foreach(KeyValuePair<int , int> p in freq){
            if (!st.Contains(p.Value))
                st.Add(p.Value);
            else 
                return false ; 
        }
        return true ; 
    }
}

File: 05- Hash Map - Set/2- Unique Number of Occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> freq;
        for (int x : arr) freq[x]++;
        unordered_set<int> st;
        for (auto& p : freq) {
            if (st.count(p.second))
                return false;
            st.insert(p.second);
        }
        return true;
    }
};
